pkg/reconciler/pipelinerun/resources/pipelineref.go

```diff
@@ -118,16 +118,32 @@ func GetVerifiedPipelineFunc(ctx context.Context, k8s kubernetes.Interface, tekt
 			source = s.URI
 		}
 		logger := logging.FromContext(ctx)
-		if config.CheckEnforceResourceVerificationMode(ctx) || config.CheckWarnResourceVerificationMode(ctx) {
-			if err := trustedresources.VerifyPipeline(ctx, p, k8s, source, verificationpolicies); err != nil {
-				if config.CheckEnforceResourceVerificationMode(ctx) {
-					logger.Errorf("GetVerifiedPipelineFunc failed: %v", err)
-					return nil, nil, fmt.Errorf("GetVerifiedPipelineFunc failed: %w: %v", trustedresources.ErrResourceVerificationFailed, err)
-				}
+		// skip the verification if no policies are found and trusted-resources-verification-no-match-policy is set to ignore
+		if len(verificationpolicies) == 0 {
```

Comment:
You may be able to roll this into verify Pipeline and return ErrNoMatchedPolicies.

Reply:
Yes! I hesitated on this for a while. The reason I return here is because I want to skip the verification early if no policies in the namespace. 
If we wait for another function to check if we could skip the verification, there may be some issues there.

Reply:
Oh I think I have a better solution. How about checking matched policies here? It looks cleaner

Reply:
I have updated this part, I have moved all the check into VerifyPipeline and VerifyTask